lib/Wikia/src/Service/User/Permissions/PermissionsServiceImpl.php

```diff
@@ -0,0 +1,391 @@
+<?php
+
+namespace Wikia\Service\User\Permissions;
+
+class PermissionsServiceImpl implements PermissionsService {
+
+	/** @var  PermissionsConfiguration */
+	private $permissionsConfiguration;
+
+	/** @var string[string] - key is user id */
+	private $localExplicitUserGroups = [];
+
+	/** @var string[string] - key is the user id */
+	private $globalExplicitUserGroups = [];
+
+	/** @var string[string] - key is the user id */
+	private $implicitUserGroups = [];
+
+	/** @var string[string] - key is user id */
+	private $userPermissions = [];
+
+	public function __construct() {
+		$this->permissionsConfiguration = new PermissionsConfigurationImpl();
```

Comment:
Hmm I guess I could do it, although this is the actual implementation of configuration which ties to the implementation of user permission, so I'm not sure it's necessary here, as this is a pair of implementations.